src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckCodeFixProvider.cs

```diff
@@ -37,22 +37,24 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             Document document, ImmutableArray<Diagnostic> diagnostics,
             SyntaxEditor editor, CancellationToken cancellationToken)
         {
-            var declaratorLocations = new HashSet<Location>();
-            foreach (var diagnostic in diagnostics)
+            using (var cleanlyRemovingEditor = editor.RemoveStatementsCleanly())
```

Comment:
it's unclear to me why you need this at all.  why can't this new behavior just be the behavior of CSharpSyntaxGenerator and VisualBasicSyntaxGenerator.RemoveNode?

Reply:
Because this is logic about removing _a number_ of nodes, not about removing _one_ node. We only know which end-of-lines to omit after all relevant nodes have been removed. So even if I went with your
> 1. Just update that specific code fix to make better syntactic edits on the tree.  this would be the simplest approach, but would limit the fix to just that improving this specific code action.

(which I am inclined to do), I would still end up with something like
```cs
using(var statmentRemovalContext = GetStatementRemovalContext(editor))
{
//...etc
}
```